ledger/internal/applications_test.go

```diff
@@ -1,365 +0,0 @@
-	kv, ok := c.stores[storeLocator{addr, aidx, global}]
-	if !ok {
-		return fmt.Errorf("no store for (%s %d %v) in mock cow", addr.String(), aidx, global)
-	}
-	kv[key] = value
-	c.stores[storeLocator{addr, aidx, global}] = kv
-	return nil
-}
-
-func (c *mockCowForLogicLedger) DelKey(addr basics.Address, aidx basics.AppIndex, global bool, key string, accountIdx uint64) error {
-	kv, ok := c.stores[storeLocator{addr, aidx, global}]
-	if !ok {
-		return fmt.Errorf("no store for (%s %d %v) in mock cow", addr.String(), aidx, global)
-	}
-	delete(kv, key)
-	c.stores[storeLocator{addr, aidx, global}] = kv
-	return nil
-}
-
-func (c *mockCowForLogicLedger) round() basics.Round {
-	return c.rnd
-}
-
-func (c *mockCowForLogicLedger) prevTimestamp() int64 {
-	return c.ts
-}
-
-func (c *mockCowForLogicLedger) allocated(addr basics.Address, aidx basics.AppIndex, global bool) (bool, error) {
-	_, found := c.stores[storeLocator{addr, aidx, global}]
-	return found, nil
-}
-
-func (c *mockCowForLogicLedger) txnCounter() uint64 {
-	return c.txc
-}
-
-func (c *mockCowForLogicLedger) incTxnCount() {
-	c.txc++
-}
-
-// No unit tests care about this yet, so this is a lame implementation
-func (c *mockCowForLogicLedger) blockHdrCached(round basics.Round) (bookkeeping.BlockHeader, error) {
-	return bookkeeping.BlockHeader{Round: round}, nil
-}
-
-func newCowMock(creatables []modsData) *mockCowForLogicLedger {
-	var m mockCowForLogicLedger
-	m.cr = make(map[creatableLocator]basics.Address, len(creatables))
-	for _, e := range creatables {
-		m.cr[creatableLocator{e.cidx, e.ctype}] = e.addr
-	}
-	return &m
-}
-
-func TestLogicLedgerMake(t *testing.T) {
-	partitiontest.PartitionTest(t)
-
-	a := require.New(t)
-
-	c := &mockCowForLogicLedger{}
-	l := newLogicLedger(c)
-	a.NotNil(l)
-	a.Equal(c, l.cow)
-}
-
-func TestLogicLedgerBalances(t *testing.T) {
-	partitiontest.PartitionTest(t)
-
-	a := require.New(t)
-
-	c := newCowMock(nil)
-	l := newLogicLedger(c)
-	a.NotNil(l)
-
-	addr1 := ledgertesting.RandomAddress()
-	ble := basics.MicroAlgos{Raw: 100}
-	c.brs = map[basics.Address]basics.AccountData{addr1: {MicroAlgos: ble}}
-	acct, err := l.AccountData(addr1)
-	a.NoError(err)
-	a.Equal(ble, acct.MicroAlgos)
-}
-
-func TestLogicLedgerGetters(t *testing.T) {
-	partitiontest.PartitionTest(t)
-
-	a := require.New(t)
-
-	addr := ledgertesting.RandomAddress()
-	aidx := basics.AppIndex(1)
-	c := newCowMock([]modsData{{addr, basics.CreatableIndex(aidx), basics.AppCreatable}})
-	l := newLogicLedger(c)
-	a.NotNil(l)
-
-	round := basics.Round(1234)
-	c.rnd = round
-	ts := int64(11223344)
-	c.ts = ts
-
-	addr1 := ledgertesting.RandomAddress()
-	c.stores = map[storeLocator]basics.TealKeyValue{{addr1, aidx, false}: {}}
-	a.Equal(round, l.Round())
-	a.Equal(ts, l.LatestTimestamp())
-	a.True(l.OptedIn(addr1, aidx))
-	a.False(l.OptedIn(addr, aidx))
-}
-
-func TestLogicLedgerAsset(t *testing.T) {
```

Comment:
@algorandskiy It was deleted in https://github.com/algorand/go-algorand/pull/4149/commits/48994e30e880afcd3cf645ba8988b3f9a6d551e5 when the associated type `mockCowForLogicLedger` was removed.

Reply:
(And that was because the whole idea of a separate logicLedger type was removed.  The LedgerForLogic interface remains, but `cowRoundState` implements it directly.

It had become a wrapper around cowRoundState with no additional data, ever since contract-to-contract work was done, where it no longer made sense to carry the appId around outside the logic package.